### PR TITLE
CI Integration

### PR DESCRIPTION
**Pull request by jameshopkins**
- [x] Run unit tests in `/application` and `/agent`
- [x] Build test agent (on `master`)
- [x] Push test agent image to ECR (on  `master`).

## Other Tasks
- [x] Bump `/agent` runtime to Node 7.2
- [x] Add Node 7.2 runtime to `/application` so we can run unit tests in CI

If anyone wants to check out the CodeFresh interface (which is *awesome* BTW), let me know and you use my creds to do so.

### CODE REVIEW

**Comment by jameshopkins** [comment]
I was going to do infra deploy us part of this story too, but I want to look into potentially more secure ways of storing AWS creds, than putting them in CodeFresh directly - since they're practically 'god mode'.

Also want to have a separate discussion about a decent integration test in CI, between the application and test agent, but this will involve some implementation change in both, so is out of scope of this PR.

File: codefresh.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
+version: '1.0'
+
+steps:
+
+  agent_test:
```
Comment (charypar) [inline codefresh.yml]: does this not have a `type`?

File: codefresh.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+      branch:
+        only: [ master ]
+
+  app_test:
+    image: node:7.2.0
+    working_directory: ${{main_clone}}/app
+    commands:
+      - npm i -g yarn
+      - npm i -g elm@0.18.0
+      - cd ui && yarn install
```
Comment (charypar) [inline codefresh.yml]: Can be just `yarn` (also to be consistent with the one above in `agent_test`).

I also believe `make test` will actually cause yarn install automatically

Comment (charypar) [inline codefresh.yml]: Actually, my bad, it doesn't, but it should - the makefile should be written in a way to trigger a build before test if necessary.

It's fairly easy to write the build step in terms of the build output (the bundles) and then make test dependent on that. Can we change? (In my mind the simpler the CI script the better and this sort of logic should be handled by the makefiles and npm scripts).

File: codefresh.yml (hunk 1 above)
Comment (jameshopkins) [inline codefresh.yml]: That's purposeful. It's an ['untyped' build step](https://docs.codefresh.io/docs/steps)

Comment (charypar) [inline codefresh.yml]: I need to do my homework on codefresh :)

**Comment by charypar** [comment]
Ran into this randomly:

> Note: There are known issues with Node.js v7 and webpack / webpack-dev-server. Please use Node.js v6 for the moment.

https://github.com/webpack/webpack-dev-server

**Comment by jameshopkins** [comment]
Ah good spot. Will roll back to a 6 branch

File: codefresh.yml (hunk 2 above)
Comment (jameshopkins) [inline codefresh.yml]: Dependency on https://github.com/httpmark/httpmark/pull/59

**Pull request merged**